src/BrowserInterop/Extensions/CallBackInteropWrapper.cs

```diff
@@ -18,9 +18,34 @@ public class CallBackInteropWrapper
 
         public bool GetJsObjectRef { get; set; }
 
+        /// <summary>
+        /// Adds javascript side debounce to the callback (in milliseconds)
+        /// see https://davidwalsh.name/javascript-debounce-function
+        /// </summary>
+        public double? Debounce { get; set; }
+
+        /// <summary>
+        /// Determines whether the callback should be invoked at the first occurrance 
+        /// or if the invocation shall be postponed until the debounce timeout is expired.
+        /// </summary>
+        public bool Immediate { get; set; }
+
+        /// <summary>
+        /// Determines whether the callback shall be invoked always when the debounce time is expired
+        /// or if it should be postponed until no event is incomming within one debounce time period.
+        /// </summary>
+        public bool TriggerPermanent { get; set; }
+
+        /// <summary>
+        /// Determines whether default values of the serialized callback data shall be transmitted
+        /// </summary>
+        public bool IncludeDefaults { get; set; }
+
+        public bool GetArgumentsSerializationAndRef { get; set; }
```

Comment:
While I understand the use case, I'm not a big fan of this being only here. Maybe we could use the serialization spec for saying we want to get a reference to the js variable (instead of "*" use "&").

Reply:
The main use case of callbacks are events I think. So mostly you might directly want the event args deserialized. But some event args also provide functions (e.g. mouse event args  => args.preventDefault()) . I think its better to provide the serialized value by default but optionally get the reference as well. I want to reduce the amount of interop calls. If I register for mouse events I don't want to call the objects data whenever the event occurs. On the other hand I might want to call the preventDefault() method, therefore I need a possibility to get both: The event data and the args instance.

Reply:
I wasn't clear sorry: I totally agree with you, this is nice to avoid 2 interop call. What I would like is to use this mechanismon method call / property get as well.

Reply:
sure that can be done. I didn't have the requirement yet.

Reply:
If I pull all the typesafe filter handling up to the client side, I could also easily create those filters you mentioned ("&" for reference "*" for serialization value)

Reply:
what do you think about using "&[]" for indicating that you want the references of array items instead of the array reference? Do you have other use cases that might it be worth to implement?

Reply:
Is it already possible today to exclude properties? For example Person =>  I want to include the whole address except the streetNumber

Reply:
"[&]" is ok, I don't mind adding more syntax to serializationSpec. We can also add person:{"*":true,"address":"!"} what do you think ?